openmc/tallies.py

```diff
@@ -304,24 +317,30 @@ def nonzero(val):
                 return 1 if not val else val
 
             # Reshape the results arrays
-            new_shape = (nonzero(self.num_filter_bins),
-                         nonzero(self.num_nuclides),
-                         nonzero(self.num_score_bins))
-
-            sum = np.reshape(sum, new_shape)
-            sum_sq = np.reshape(sum_sq, new_shape)
+            sum = np.reshape(sum, self.shape)
+            sum_sq = np.reshape(sum_sq, self.shape)
```

Comment:
Oh I see, yes this particular change is my own. I'll remove `nonzero` here. But much of the diff below is from #525 and not my own code changes for this PR.